itchiodl/library.py

```diff
@@ -40,39 +41,67 @@ def load_owned_games(self):
                 break
             page += 1
 
+    def load_game_page_keys(self, page):
+        """Load a page of game keys via the API"""
+        duplicates = 0
+        print("Loading page", page)
+        r = requests.get(
+            f"https://api.itch.io/profile/owned-keys?page={page}",
+            headers={"Authorization": self.login},
+        )
+        j = json.loads(r.text)
+
+        for s in j["owned_keys"]:
+            if str(s.get("game_id")) in self.key_pairs:
+                duplicates+=1
+                # The duplicate download key check assumes that a consecutive
+                # string of ten identical game ids indicates the list has not changed
+                # Duplicate game keys can happen if you buy multiple bundles containing the same item
+                # print("duplicate game ID", s["game_id"]) # Old debug line
+                if duplicates > 9 and exists("key_pairs.json"):
```

Comment:
I don't understand why you're testing for the file existing. If the file exists, you'll have loaded it, and if it doesn't you won't have duplicate keys

Reply:
Good point, duplicate keys naturally occur in a first read through any large enough library that's bought multiple bundles but it'd be weird to buy 10 copies of the same title in a row, theoretically possible but in practice nobody will